nidx/nidx_relation/src/graph_query_parser.rs

```diff
@@ -482,42 +485,52 @@ impl<'a> GraphQueryParser<'a> {
             tokenized_terms.push(tantivy::Term::from_field_text(tokenized_field, &token.text));
         }
 
-        // TODO: Rethink this
-        // Current logic:
-        // - Exact match always match the search term against the full field
-        // - Fuzzy + prefix search works does a prefix fuzzy match of the whole entity name
-        // - Fuzzy search looks for entities containing all words in the term with a fuzzy match (tokenized)
-        //
-        // Questions:
-        // - Do we want exact match of a word in the entity? (kind of supported by setting fuzzy distance = 0)
         let query: Box<dyn Query> = match value {
             Term::Exact(_) => Box::new(TermQuery::new(exact_term, IndexRecordOption::Basic)),
-            Term::Fuzzy(fuzzy) => match fuzzy {
-                FuzzyTerm {
-                    fuzzy_distance,
-                    is_prefix: true,
-                    ..
-                } => Box::new(FuzzyTermQuery::new_prefix(exact_term, *fuzzy_distance, true)),
-
-                FuzzyTerm { fuzzy_distance, .. } => {
-                    if tokenized_terms.len() > 1 {
+
+            Term::ExactWord(_) => Box::new(TermSetQuery::new(tokenized_terms)),
+
+            Term::Fuzzy(fuzzy) => {
+                if fuzzy.is_prefix {
+                    Box::new(FuzzyTermQuery::new_prefix(exact_term, fuzzy.fuzzy_distance, true))
+                } else {
+                    Box::new(FuzzyTermQuery::new(exact_term, fuzzy.fuzzy_distance, true))
+                }
+            }
+
+            Term::FuzzyWord(fuzzy) =>
+            {
+                #[allow(clippy::collapsible_else_if)]
+                if fuzzy.is_prefix {
```

Comment:
You can remove a bunch of duplication here by extracting the function to a variable. Something like:

```
let query_builder = if fuzzy { FuzzyTermQuery::new_prefix } else { FuzzyTermQuery::new };
```

and then use it in a single codepath like:
```
  if tokenized_terms.len() == 1 {
      let tokenized_term = tokenized_terms.into_iter().next().unwrap();
      Box::new(query_builder(tokenized_term, fuzzy.fuzzy_distance, true))
  } else {
```

Reply:
Done :)